packages/db/native/db/dump.zig

```diff
@@ -48,9 +53,21 @@ pub fn loadCommon(napi_env: c.napi_env, info: c.napi_callback_info) callconv(.C)
     const ctx = napi.get(*db.DbCtx, napi_env, args[1]) catch return null;
     const errlog = napi.get([]u8, napi_env, args[2]) catch return null;
 
+    var com: selva.selva_dump_common_data = .{
+        .errlog_buf = errlog.ptr,
+        .errlog_size = errlog.len,
+    };
+    const rc = selva.selva_dump_load_common(ctx.selva, &com, sdb_filename.ptr);
     var res: c.napi_value = null;
-    const rc = selva.selva_dump_load_common(ctx.selva, sdb_filename.ptr, errlog.ptr, errlog.len);
     _ = c.napi_create_int32(napi_env, rc, &res);
+
+    if (com.meta_data != null) {
+        const ptr: [*]u32 = @ptrCast(@alignCast(@constCast(com.meta_data)));
+        const len = com.meta_len / @sizeOf(u32);
+        ctx.ids = ctx.allocator.dupe(u32, ptr[0..len]) catch return null;
+        selva.selva_free(@constCast(com.meta_data));
```

Comment:
You should always free the returned buffer, hence:
```suggestion
        const ptr: [*]u32 = @ptrCast(@alignCast(@constCast(com.meta_data)));
        const len = com.meta_len / @sizeOf(u32);
        defer selva.selva_free(@constCast(com.meta_data));
        ctx.ids = ctx.allocator.dupe(u32, ptr[0..len]) catch return null;
```